everything_else/reproducing.m

```diff
@@ -269,9 +269,48 @@
 RHS=min(reshape(b1.*b2 + ba1 + 2*ba2 + 2*(1-b1).*(1-ba1) + 2*(1-b2).*(1-ba1) + 2*(1-b3).*(1-ba2) + 2*(1-ba2).*(1-ba1) - 2*(1-b3) - 1 + b1.*b3 - b2,4,[]));
 isequal(LHS,RHS);
 
-%% Pg. 30, PTR-CZW
-%% Pg. 31, Bit-flipping
-%% Pg. 32, SFR-ABCG-1
+%% PG 30
+z=[1 0; 0 -1];
+z1 = kron(z,eye(16));
+z2 = kron(kron(eye(2),z),eye(8));
+z3 = kron(kron(eye(4),z),eye(4));
+z4 = kron(kron(eye(8),z),eye(2));
+%% Alternatively, zlist = zgenerate(4);
+LHS=z1*z2*z3*z4;
+b=bgenerate(4);
```

Comment:
I appreciate the value of having a function like `bgenerate` but throughout this whole code which has been the result of collaborative contributions involving dozens of people, we have just been using `dec2bin(2^4-1 :-1 : 0)-'0'` instead of something like `bgenerate(4)`. An advantage of sticking to the former, is that it will run in anyone's MATLAB or Octave console without requiring them to go and find what `bgenerate` does.

Reply:
Would you like me to remove all of my custom functions?